WordPress/src/main/java/org/wordpress/android/ui/main/WPMainActivity.java

```diff
@@ -248,6 +263,23 @@ public void onPageScrolled(int position, float positionOffset, int positionOffse
         // We need to register the dispatcher here otherwise it won't trigger if for example Site Picker is present
         mDispatcher.register(this);
         EventBus.getDefault().register(this);
+
+        if (authTokenToSet != null) {
+            // Save Token to the AccountStore. This will trigger a onAuthenticationChanged.
+            AccountStore.UpdateTokenPayload payload = new AccountStore.UpdateTokenPayload(authTokenToSet);
+            mDispatcher.dispatch(AccountActionBuilder.newUpdateAccessTokenAction(payload));
+        }
+    }
+
+    private boolean hasMagicLinkLoginIntent() {
+        String action = getIntent().getAction();
+        Uri uri = getIntent().getData();
+
+        return Intent.ACTION_VIEW.equals(action) && uri != null && uri.getHost().contains(SignInActivity.MAGIC_LOGIN);
+    }
+
+    private String getAuthToken() {
+        return getIntent().getData().getQueryParameter(SignInActivity.TOKEN_PARAMETER);
```

Comment:
I think we need a null check on `getData()` here. Right now it's not a problem since `hasMagicLinkLoginIntent()` does the null check before this is called, but I can see this being called elsewhere down the road.